wcomponents-core/src/test/java/com/github/bordertech/wcomponents/AbstractWComponentTestCase.java

```diff
@@ -147,6 +148,95 @@ protected void assertAccessorsCorrect(final WComponent component, final String m
 		}
 	}
 
+	/**
+	 * This method checks that the a component model uses the default model on creation of a component.
+	 * @param wComponent the component to test the model
+	 */
+	protected void assertComponentModelUsesDefaultOnCreation(AbstractWComponent wComponent) {
+		ComponentModel shared = wComponent.getDefaultModel();
+		wComponent.setLocked(true);
+		ComponentModel model = wComponent.getOrCreateComponentModel();
+		org.junit.Assert.assertSame(model, shared);
+	}
+
+	/**
+	 * This method checks that the a component model uses the default model when the same value is set as the same in
+	 * the default model, rather than creating a new model.
+	 * @param wComponent the component the model is being created from
+	 * @param method the method used to change the model
+	 * @param userContextValue the value to be used within the user context
+	 * @param setterArgs array matching the variable argument type
+	 */
+	protected void assertComponentModelUsesDefaultOnSameValue(AbstractWComponent wComponent, String method,
+															  Object userContextValue, Object setterArgs[]) {
+		wComponent.setLocked(false);
+		invokeSetMethod(wComponent, method, userContextValue, setterArgs);
+		ComponentModel shared = wComponent.getDefaultModel();
+		wComponent.setLocked(true);
+
```

Comment:
Yes - just a quick test that `getOrCreateComponentModel` is indeed getting the default model instead of creating a new one (without an active context).